src/storage/on_disk.rs

```diff
@@ -0,0 +1,113 @@
+use std::path::Path;
+
+use alloy_primitives::{keccak256, Address, Bytes, FixedBytes, B256, B64, U256};
+use libmdbx::{Database, DatabaseOptions, NoWriteMap};
+use revm::primitives::Bytecode;
+
+use super::{AccountBasic, EvmCode, Storage};
+
+/// A storage that stores chain data in a MDBX database.
+#[derive(Debug)]
+pub struct OnDiskStorage {
+    inner: Database<NoWriteMap>,
+}
+
+impl OnDiskStorage {
+    /// Opens the on-disk storage at the specified path.
+    pub fn open(path: impl AsRef<Path>) -> Result<Self, libmdbx::Error> {
+        let db = Database::open_with_options(
```

Comment:
Will `open` work for now?

Reply:
No. We need to specify `max_tables`.

```
Err(ExecutionError("database error: MDBX_DBS_FULL: Too many DBI-handles (maxdbs reached)"))
```

Reply:
We should take in the options then since the callee knows these limits the best. Hardcoding anything here will only reduce usage.

Reply:
> the callee knows these limits the best

`OnDiskStorage` is assuming that the table names are `balance`, `nonce`, `code_hash`, etc. So I think it is safe to assume that `OnDiskStorage` already know what tables there are and how many tables there are. So I think `OnDiskStorage` should dictate the param `max_tables` instead of letting callees specifying it.

Reply:
No. A `trait` requires the implementer to has some signatures but makes no assumption about the actual `struct`ure or implementation. Similarly, `OnDiskMemory` only has some requirements on some tables, but generally  `db: Database<NoWriteMap>` accepts any database schema. We cannot assume that, say, an Reth ExEx that uses `pevm` would have fewer tables, or force them to use different databases for different things.